### PR TITLE
Use STAC datacube extension

### PR DESCRIPTION
**Pull request by forman**
Collection items now make use the STAC [datacube](https://github.com/stac-extensions/datacube) extension.

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* [ ] ~Add docstrings and API docs for any new/modified user-facing classes and functions~
* [ ] ~New/modified features documented in `docs/source/*`~
* [x] Changes documented in `CHANGES.md`
* [x] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

File: xcube/webapi/ows/stac/controllers.py [hunk 1]
```diff
@@ -302,20 +312,21 @@ def _get_dataset_feature(ctx: DatasetsContext,
...
     # TODO: Prefer original storage location.
     #       The "s3" operation is default.
     default_storage_url = f"{base_url}/s3/datasets"
 
     if 'time' in dataset:
-        start_time = _format_timestamp(dataset['time'][0].values)
-        end_time = _format_timestamp(dataset['time'][-1].values)
+        time_var = dataset['time']
+        start_time = to_json_value(time_var[0])
+        end_time = to_json_value(time_var[-1])
```
Comment (pont-us) [inline xcube/webapi/ows/stac/controllers.py:329]: Can we guarantee that these will always be datetime64s and not e.g. plain datetimes or pandas Timestamps?

File: xcube/webapi/ows/stac/controllers.py [hunk 2]
```diff
@@ -342,10 +353,12 @@ def _get_dataset_feature(ctx: DatasetsContext,
             ],
         },
         "properties": {
-            "xcube:dimensions": dict(dataset.dims),
-            "xcube:variables": variables,
-            "xcube:coordinates": coordinates,
-            "xcube:attributes": dict(dataset.attrs),
+            "datacube:dimensions": dc_dimensions,
+            "datacube:variables": dc_variables,
```
Comment (pont-us) [inline xcube/webapi/ows/stac/controllers.py:357]: Why `datacube` and not `cube`? This won't validate against the datacube schema.

File: test/webapi/ows/stac/stac-item.json [hunk 3]
```diff
@@ -0,0 +1,1150 @@
...
+        ],
+        [
+          0.0,
+          50.0
+        ]
+      ]
+    ]
+  },
+  "properties": {
+    "datacube:dimensions": {
```
Comment (pont-us) [inline test/webapi/ows/stac/stac-item.json:42]: Datacube schema specifies `cube:`, not `datacube:` -- typo?

File: test/webapi/ows/stac/stac-item.json [hunk 4]
```diff
@@ -0,0 +1,1150 @@
...
+      "time": {
+        "type": "temporal",
+        "values": [
+          "2017-01-22T00:00:00Z",
+          "2017-01-29T00:00:00Z",
+          "2017-02-05T00:00:00Z"
+        ]
+      }
+    },
+    "datacube:variables": {
```
Comment (pont-us) [inline test/webapi/ows/stac/stac-item.json:76]: Datacube schema specifies `cube:`, not `datacube:` -- typo?

**Review by pont-us — CHANGES_REQUESTED** [review]
Just a couple of queries but mostly fine.

File: xcube/webapi/ows/stac/controllers.py (hunk 1 above)
Comment (forman) [reply xcube/webapi/ows/stac/controllers.py:329]: No. We can also not guarantee that Pandas timestamps will stay the same. However, xarray time coordinates are `np.datetime64`.

File: xcube/webapi/ows/stac/controllers.py (hunk 2 above)
Comment (forman) [reply xcube/webapi/ows/stac/controllers.py:357]: Argh, I overlooked the most obvious thing! Will change...

File: test/webapi/ows/stac/stac-item.json (hunk 3 above)
Comment (forman) [reply test/webapi/ows/stac/stac-item.json:42]: Yes. Stupidity.

File: test/webapi/ows/stac/stac-item.json (hunk 4 above)
Comment (forman) [reply test/webapi/ows/stac/stac-item.json:76]: See above.

File: xcube/webapi/ows/stac/controllers.py (hunk 1 above)
Comment (pont-us) [reply xcube/webapi/ows/stac/controllers.py:329]: OK. If it's enough for our current requirements I guess we can leave it for now and add more handlers to `jsonencoder` if and when the need arises.

Comment (forman) [reply xcube/webapi/ows/stac/controllers.py:329]: However, we may enhance `to_json_value()` to also handle Pandas timestamps explicitly.

**Review by pont-us — APPROVED** [review]
Looks good!

**Pull request merged**